public/app/plugins/datasource/elasticsearch/components/QueryEditor/BucketAggregationsEditor/SettingsEditor/DateHistogramSettingsEditor.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { ComponentProps, useState } from 'react';
```

Comment:
This file is change a refactoring of public/app/plugins/datasource/elasticsearch/components/QueryEditor/BucketAggregationsEditor/SettingsEditor/index.tsx. 

the fix for the creatable option being case sensitive is in 76b27dd8bbb8b48b7c7410614955b601c5e7efca

plus, here i did some changes to handle persistence of custom options in the UI as it wasn't handled before)